src/aibtc-dao/extensions/payments-invoices/public/pay-invoice.ts

```diff
@@ -69,6 +71,16 @@ async function main() {
     CONFIG.ACCOUNT_INDEX
   );
   const nextPossibleNonce = await getNextNonce(CONFIG.NETWORK, address);
+
+  // Add post-conditions to ensure sender has approved sufficient funds
+  // Note: The actual amount and token type should be read from the invoice
+  // This is a placeholder that ensures the transaction will fail if post-conditions aren't met
+  const postConditions = [
+    Pc.principal(address)
+      .willSendEq(0) // Will be replaced with actual amount from invoice
```

Comment:
Same here with `get-invoice` in the contract
https://github.com/aibtcdev/aibtcdev-contracts/blob/02470aa88060def8f11bbd8e05e78262eefe0f90/contracts/dao/extensions/aibtc-payments-invoices.clar#L355-L358